index.js

```diff
@@ -111,6 +116,51 @@ Babel.prototype.optionsHash = function() {
       hash[key] = (typeof value === 'function') ? (value + '') : value;
     }
 
+    if (options.plugins) {
+      hash.plugins = [];
+
+      for (var i = 0; i < options.plugins.length; i++) {
+        var plugin = options.plugins[i];
+
+        var pluginType = typeof plugin;
+        var pluginAugmentsCacheKey = false;
+        var pluginProvidesBaseDir = false;
+        var pluginRequiresBaseDir = true;
+
+        // do not look for properties on string plugins
+        if (Array.isArray(plugin)) {
+          // when an array, the second argument is an options hash
+          pluginRequiresBaseDir = false;
+          hash.plugins.push(stringify(plugin[1]));
```

Comment:
Yes, that is my understanding.  But to be clear this is for basic babel 6 support, but I am not 100% sure what is allowed and disallowed in the options provided to a plugin. 

I'm happy to follow this up with a recursive algo that does the same treatment to each key in this second arg also, if we think that is needed.

In the meantime, for babel 5 support, we don't even hit this branch.